### PR TITLE
ISPN-7536 JDBCMigrator is unable to read 8.x tables due to marshaller…

### PR DESCRIPTION
**Pull request by ryanemerson**
… changes.

- A stripped down version of the 8.x StreamingMarshaller implemented in org.infinispan.tools as LegacyVersionAwareMarshaller.
- LegacyVersionAwareMarshaller is restricted to unmarshalling.
- Tests created that utilise a .bin file generated from Infinispan 8.x to ensure that 8.x bytes can be succesfully unmarshalled.
- Migration guide updated to explicitly state the impact of marshalling changes on persistence stores.

https://issues.jboss.org/browse/ISPN-7536

### CODE REVIEW

File: tools/src/test/java/org/infinispan/tools/jdbc/migrator/marshaller/LegacyVersionAwareMarshallerTest.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+   public LegacyVersionAwareMarshallerTest() {
+      Map<Integer, AdvancedExternalizer<?>> externalizerMap = new HashMap<>();
+      externalizerMap.put(256, new TestObjectExternalizer());
+      marshaller = new LegacyVersionAwareMarshaller(externalizerMap);
+   }
+
+   @BeforeTest
+   public void beforeTest() throws Exception {
+      Path path = new File("src/test/resources/marshalled_bytes_8.x.bin").toPath();
```
Comment (gustavonalle) [inline tools/src/test/java/org/infinispan/tools/jdbc/migrator/marshaller/LegacyVersionAwareMarshallerTest.java]: Can you add a script or write some directions on how this .bin is generated?

**Comment by gustavonalle** [comment]
@ryanemerson LGTM, just a small comment

File: tools/src/test/java/org/infinispan/tools/jdbc/migrator/marshaller/LegacyVersionAwareMarshallerTest.java (hunk 1 above)
Comment (ryanemerson) [inline tools/src/test/java/org/infinispan/tools/jdbc/migrator/marshaller/LegacyVersionAwareMarshallerTest.java]: @gustavonalle I've added the sample program and instructions as a comment at the bottom of this class.

**Comment by gustavonalle** [comment]
@ryanemerson integrated, thanks!

**Pull request merged**